Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -2913,6 +2927,42 @@ bool RenderLayerBacking::isDirectlyCompositedImage() const
     return false;
 }
 
+bool RenderLayerBacking::isUnscaledBitmapOnly() const
+{
+    if (!is<RenderImage>(renderer()) && !is<RenderHTMLCanvas>(renderer()))
+        return false;
+
+    // FIXME: Make sure we handle invalidation correctly, disabling the changes
+    // if we add decorations (or resize the image?)
+    if (m_owningLayer.hasVisibleBoxDecorationsOrBackground())
+        return false;
+
+    if (is<RenderImage>(renderer())) {
+        // FIXME: Test image path.
+        auto& imageRenderer = downcast<RenderImage>(renderer());
+        if (auto* cachedImage = imageRenderer.cachedImage()) {
+            if (!cachedImage->hasImage())
+                return false;
+
+            auto* image = cachedImage->imageForRenderer(&imageRenderer);
+            if (!is<BitmapImage>(image))
+                return false;
+
+            if (downcast<BitmapImage>(*image).orientationForCurrentFrame() != ImageOrientation::None)
+                return false;
+
+            // If we have a hidpi image (RenderImage::imageDevicePixelRation), is that accounted for here?
+            return contentsBox().size() == image->size();
```

Comment:
Probably need some snapped rect comparisons here, and a test with a composited image at a sub-pixel offset. If we get this wrong, images might jiggle when compositing status changes.